### ISSUE TITLE
Gerald needs a new tube

### ISSUE THREAD

**Issue by amcewen** [DoESLiverpool/somebody-should#1427]
Gerald's cutting power dropped drastically yesterday, @Sean-anotherone and @JackiePease investigated and it seems it's the tube.  (I has been [a while since we bought our last one](https://github.com/DoESLiverpool/somebody-should/issues/987), and it's been running lots, and over max power, of late)

@JackiePease contacted HPC and has driven over there to pick up [a new tube](http://hpclaser.co.uk/index.php?main_page=product_info&cPath=2&products_id=7)

**Comment by Sean-anotherone** [comment]
Thanks to @JackiePease driving to pick up the tube (from a supplier that opened on a Sunday to allow it to be picked up) Gerald is now optically as near to perfectly set up as it can be - machine is cutting really well, tiny kerf and really clean cuts.

Of note, when the beam was just a tiny bit off centre in the wexit aperture (about 1.5mm away from being perfect) the machine was behaving really strangely - cutting horizontally really well, but cutting vertically poorly.  This was really strange but readjusting the final mirror just a tiny bit fixed it.

The focus tool is no good for setting the Z on plastic sheets - it needs to be set manually by looking at the hole size.

The remaining problem on the machine is something to do with the software (the toolpath to be precise).  At a quite predictable point in the program the machine "clunks" and skips a few teeth on the vertical drive, not while cutting, but wile transiting to the next row - ruining the sheet.  The problem was very repeatable but @amcewen and I spent about 2hrs trying everything we could think of before Adrian came up with a *possible* answer.

By amending the drawing so the offending part of the cut did not involve a transit between rows, the problem seemed to go away, when I left, @amcewen was in the process of manually editing the cut order for the entire sheet.

**Comment by amcewen** [comment]
Turns out my amending the drawing didn't make any difference, @Sean-anotherone, or rather just moved the problems elsewhere.  However, I did find and fix the skipping problem!  The speed setting on the LCD screen on Gerald was set to 89%, and pushing that up to 100% sorted out the skipping issue.

It didn't quite cut a full sheet properly, but I think that's more down to getting the focus and speed settings right.

**Comment by amcewen** [comment]
@JackiePease spoke to the guy at HPC, who reckoned it's unlikely to be the power supply, as they either work or don't.

Power does seem to fade during the cut though, by the time it gets to the black layer in the current cut file (after ~12 minutes) there's not enough power to mark the material.  The laser is still outputting, as I get a tiny hole at the end of the path if I pause it around that point.

I was wondering if something was overheating, so have paused it and left it for an hour, and that made no difference to the output.  Stopping it and then restarting the cut straight away it was cutting fine, so it doesn't look like something that's affected by the amount of time the laser has been running (i.e. the power supply fading after lots of use, or the lens or mirrors heating up or something)

I tried removing the air assist cowl and that had no effect, so if it's misaligned it's before it gets to that.

Trying a test cut on top of one of the cuts that hardly marked the surface (within a mm of the cut) and it cut through beautifully, so it's not just a case of the position on the bed.

Running short on ideas, but continuing to investigate...

**Comment by amcewen** [comment]
I seem to have found the problem.

The cut file for the visors has three layers: one with the internal cuts; a low power one to write the version numbers onto the visors; and a final one with the external cuts.

For some reason the software seems to miss the command to up the power when it moves from the low power to the final layer, so all the external cuts are at far too low a power.  Omitting the low power layer means it makes all the cuts just fine.  I've moved it to the end of the sequence now, so hopefully we'll still get version numbers without interfering with the cut.

There are a couple of points on the bed where it's not quite cutting perfectly.  @Sean-anotherone is coming in later to do some final focus/alignment checks, so I'll leave this open till after that.

**Comment by johnmckerrell** [comment]
That's quite an insidious little bug! I wonder if we've had that before at other times that have resulted in complaints that Gerald's not cutting properly, only for others to think it's fine.

**Comment by ajlennon** [comment]
Geez. Nightmare.

**Comment by Sean-anotherone** [comment]
I did a whole raft of optimisation tests and found something really odd.  The focus-depth has changed from the ~19mm is has been for *ever* to ~26mm.
I have made a new tool to set the distance between the end of the output cone and the sheet of material.
Also, the poly cuts better at power 50 speed 70 than it does with either higher power or lower speed or any other combination, and I tried lots of combinations. Go figure.  I guess the higher power is remelting the sheet back together.
I left after cutting a full sheet that needed minimal "picking - the parts were separate and all cuts were right through, it took 20 mins.
It will be necessary to clean out the bed regularly or all the little bits that drop in will make a nice bonfire.

I also managed to leave my measuring loupe in the event room, would appreciate if someone could keep it safe for me.
Sean


    On Monday, 20 April 2020, 17:11:40 BST, John McKerrell <notifications@github.com> wrote:  
 
 


That's quite an insidious little bug! I wonder if we've had that before at other times that have resulted in complaints that Gerald's not cutting properly, only for others to think it's fine.
